client_plugin/drivers/shared/shared_driver.go

```diff
@@ -116,35 +119,48 @@ func NewVolumeDriver(cfg config.Config, mountDir string) *VolumeDriver {
 		d.internalVolumeDriver = *internalVolumeParam
 	}
 
+	// Use go routine due to the timeout for plugin initialization
+	go d.backgroundInitTasks()
+
+	log.WithFields(log.Fields{
+		"version": version,
+	}).Info("vSphere shared plugin started ")
+	return &d
+}
+
+// backgroundInitTasks: create new dockerOps, load server image, and start key-value store
+func (d *VolumeDriver) backgroundInitTasks() {
 	// create new docker operation client
 	d.dockerOps = dockerops.NewDockerOps()
 	if d.dockerOps == nil {
 		log.Errorf("Failed to create new DockerOps")
-		return nil
+		return
 	}
 
 	// Load the file server image
 	go d.dockerOps.LoadFileServerImage()
 	log.Infof("Started loading file server image")
 
 	// initialize built-in etcd cluster
-	etcdKVS := etcdops.NewKvStore(d.dockerOps)
-	if etcdKVS == nil {
-		log.Errorf("Failed to create new KV store")
-		return nil
+	for {
```

Comment:
Do we retry infinitely here? I think we should have a limited retry count.

Reply:
No it's the same as before the change, if the initialization failed, the plugin will always try to restart. Similarly, we should forever try to start the ETCD, until the plugin is disabled.